### ISSUE TITLE
Add pre-processing callback that caches the result

### ISSUE THREAD

**Issue by JustYourAverageUTFan** [JujuAdams/Scribble#555]
Currently, support to add custom commands that can be called in-string does exist, but what if I wanted to attach a specific function to a character itself? Sort of like how you can make `#` render a newline if a specific macro is set to true.

What I'm trying to say is that there should be the option to call some sort of function, attach it to a character, and tell Scribble to execute some sort of function when that character is inserted in the text - kind of like a mini-command.

```game-maker-language
//for example
scribble_character_attach_event("*", instance_create_depth(10,78,-200,obj_example));
```

Thank you for your time.
While this probably doesn't exist yet, is there another method of doing this?

**Comment by JujuAdams** [comment]
When would you want the "attached event" to be executed? Bear in mind that Scribble text is conventionally drawn every frame.

**Comment by JustYourAverageUTFan** [comment]
Just like commands, I think. What I mean is something like every "x" being yellow, for example, or something more complex like $ being able to insert a variable

**Comment by JujuAdams** [comment]
I won't be adding the feature as you specifically requested, but I can definitely see the utility in a cached preprocessing step. This will look like:

```gml
var _string = "This will cost you $100!";

scribble(_string)
.preprocess(function(_string) { return string_replace_all(_string, "$", "[c_yellow]$[/c]"); })
.draw(x, y);
```

I will add a global default preprocessing function setter too.

**Comment by JustYourAverageUTFan** [comment]
Thanks!

**Comment by JujuAdams** [comment]
Implemented for Scribble 9.1.0
